src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -946,6 +947,13 @@ def do_resolve_with_resolved_children(
                 logical_plan,
             )
 
+        if isinstance(logical_plan, Drop):
```

Comment:
Is SelectStatement itself good enough? I'm thinking of removing this new logical plan (and Rename).

Reply:
I see that nearly every API in dataframe.py checks if `self._select_statement` is None, and if it is, the API code defaults to using a logical plan (saw this in order by, sort, filter, etc). The `Drop` plan is used in a similar way.